unitex/src/fr/umlv/unitex/leximir/helper/DelacHelper.java

```diff
@@ -0,0 +1,391 @@
+import java.util.List;
+import fr.umlv.unitex.leximir.model.Delac;
+import fr.umlv.unitex.leximir.model.DictionaryPath;
+import fr.umlv.unitex.leximir.util.Utils;
+
+/**
+ * This class is an helper for delac entry
+ * @author Rojo Rabelisoa
+ */
+public class DelacHelper {
+    /**
+     * This function return a list of dictionary in directory
+     * @return return an ArrayList of dictionary
+     * @throws FileNotFoundException if there are no dictionary found in StaticValue.allDelc path 
+     */
+    public static ArrayList<String> getDicDelacPath() throws FileNotFoundException, IOException{
+        ArrayList<String> list= new ArrayList<>();
+        File folder = new File(DictionaryPath.allDelac);
+        File[] listOfFiles = folder.listFiles();
+        for (File file : listOfFiles) {
+            if (file.isFile()) {
+                if (file.getName().endsWith("dic")) {
+                    list.add(file.getName());
+                }
+            } 
+        }
+        if(list.isEmpty()){
+            throw new FileNotFoundException("dictionary not found");
+        }
+        return list;
+    }
+    
+    /**
+     * This function get all dictionaries in delac folder and return an Object [][] which contains all information(
+     * POS, lemmaAll, lemma, fSTCode, SynSem, comment, wn_SinSet, lemmaId, dicFile)
+     * @param alldelac if alldelac is true, the function takes all delas in delcs folder, else it takes dictionary selected in configuration
+     * @return List of lemma in Object[][] format
+     * @throws FileNotFoundException
+     * @throws IOException 
+     */
+    public static Object[][] getAllDelacFromDicToObject(boolean alldelac, File dic) throws FileNotFoundException, IOException{
+        List<String> list= new ArrayList<>();
+        if(alldelac){
+            list = getDicDelacPath();
+        }
+        else{
+            list=Arrays.asList(dic.getAbsolutePath().toString());
+        }
+        Delac delac = new Delac();
+        Field[] lf = delac.getClass().getDeclaredFields();
+        int count =0;
+        for(String dela:list){
+            //String path = Utils.getValueXml("pathDelas")+"/"+dela;
+            String path="";
+            if(alldelac)
+            path = DictionaryPath.allDelac+"//"+dela;
+            else
+            path = dela;
+            
+            ArrayList<String> readFile = Utils.readFile(path);
+            for(String s:readFile){
+                if(s.trim().length()>0) {
+                    count++;
+                }
+            }
+            DictionaryPath.dictionary.add(dela);
+        }
+        
+        Object[][] ob = new Object[count][lf.length];
+        int k=0;
+        int lemmaId=0;
+        for(String dela:list){
+            String pOs,lemmaAll,lemma,fSTCode,SynSem,comment,wn_SinSet;
+            
+            String dicFile=dela;
+            String path="";
+            if(alldelac)
+            path = DictionaryPath.allDelac+"//"+dela;
```

Comment:
Do not hardcode path separators, use instead  `File.pathSeparatorChar` and do not forget to use [FileUtil](../../tree/master/unitex/src/fr/umlv/unitex/files/FileUtil.java) functions when necessary.

Reply:
Done ! but I used File.separator, because File.pathSeparatorChar is used for PATH environment variable